datafusion/expr/src/logical_plan/invariants.rs

```diff
@@ -109,12 +210,13 @@ pub fn check_subquery_expr(
         match outer_plan {
             LogicalPlan::Projection(_)
             | LogicalPlan::Filter(_)
+            | LogicalPlan::TableScan(_)
             | LogicalPlan::Window(_)
             | LogicalPlan::Aggregate(_)
             | LogicalPlan::Join(_) => Ok(()),
             _ => plan_err!(
                 "In/Exist subquery can only be used in \
-                Projection, Filter, Window functions, Aggregate and Join plan nodes, \
+                Projection, Filter, TableScan, Window functions, Aggregate and Join plan nodes, \
```

Comment:
The `push_down_filter` optimization pass can push a `IN(<subquery>)` into a TableScan. This change was required for [push_down_filter::tests::test_in_subquery_with_alias](https://github.com/apache/datafusion/blob/452a8f4fa8fbfd88a5a71310bc28b9f78685c66f/datafusion/optimizer/src/push_down_filter.rs#L2836) to pass.